### PR TITLE
Add `/api/feature_flags/user_status?distinct_id=ID`

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- This adds an API endpoint which an authenticated user can use to get the active feature flags for any distinct ID.
- The endpoint is: `/api/feature_flags/user_status?distinct_id=ID`
- The idea/request came from the [PostHog users Slack](https://posthogusers.slack.com/archives/C01FKPGG5U6/p1624118681084500)
- This might even be handy for the toolbar as well.

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [ ] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

**Comment by mariusandra** [comment]
I added two more changes, now ready for a look.

1. The endpoint is actually `/api/feature_flags/user_status?distinct_id=ID` (and not `active_flags` as it was before, I edited it out to not be confusing).
2. The endpoint returns an object: 
```
{ distinct_id: 'echos back so you can verify', enabled_flags: ['flag1', 'flag2'] }
```

Previously I returned just an array, but that would have made this hard to extend, e.g. when we add feature flag values.

File: posthog/api/feature_flag.py [hunk 1]
```diff
@@ -106,3 +109,9 @@ def get_queryset(self) -> QuerySet:
         if self.action == "list":
             queryset = queryset.filter(deleted=False)
         return queryset.order_by("-created_at")
+
+    @action(methods=["GET"], detail=False)
+    def user_status(self, request: request.Request, **kwargs):
+        distinct_id = request.GET.get("distinct_id", None)
+        flags = get_active_feature_flags(self.team, distinct_id)
+        return Response({"distinct_id": distinct_id, "flags_enabled": flags})
```
Comment (paolodamico) [inline posthog/api/feature_flag.py:117]: Not a deal breaker, but I think it's worth using a serializer here to handle this logic (and reading the `GET` params too), it's cleaner and standardized. If you do keep it this way, probably worth switching to `JsonResponse` instead.

File: posthog/api/feature_flag.py [hunk 2]
```diff
@@ -106,3 +109,9 @@ def get_queryset(self) -> QuerySet:
         if self.action == "list":
             queryset = queryset.filter(deleted=False)
         return queryset.order_by("-created_at")
+
+    @action(methods=["GET"], detail=False)
+    def user_status(self, request: request.Request, **kwargs):
```
Comment (paolodamico) [inline posthog/api/feature_flag.py:114]: Also not a big deal, but `user_status` didn't seem to explain well what this endpoint does. Perhaps `for_user`? Unsure about the best semantics here.

**Review by paolodamico — APPROVED** [review]
This is super helpful! Some minor comments, but lgtm otherwise. Reminder to check mypy before merging first.

File: posthog/api/feature_flag.py (hunk 2 above)
Comment (mariusandra) [reply posthog/api/feature_flag.py]: Renamed to "for_user", though I'm also not sure what's the best name/convention here.

File: posthog/api/feature_flag.py (hunk 1 above)
Comment (mariusandra) [reply posthog/api/feature_flag.py:119]: Hmm. I chose `Response` just so we could have the nicer looking Django API like on ~~all~~ most other endpoints:

image

... yet it should still work if you ask for the json programmatically:

image

As for moving this to a serializer. I'm willing to do it if it makes sense. Yet with keeping things simple, a 3-line function body seems to have less overhead than making another class. Had it been an normal API endpoint that requires some serialization and sanitization, I'd definitely gone for it, but now for a custom JSON endpoint it feels like overkill. I might be wrong though --> I don't have muscle memory for django best practices, yet :).

File: posthog/api/feature_flag.py (hunk 2 above)
Comment (paolodamico) [reply posthog/api/feature_flag.py]: Yeah not a huge fan either, but probably not worth spending any more time on this.

File: posthog/api/feature_flag.py (hunk 1 above)
Comment (paolodamico) [reply posthog/api/feature_flag.py:119]: Yeah this seems fine. Also checked, and the `Content-Type` is properly set, so 👍

**Review by paolodamico — APPROVED** [review]
🚢

**Pull request merged**